### ISSUE TITLE
Non-existing files are visible in a folder

### ISSUE THREAD

**Issue by Aga-C** [SimpleMobileTools/Simple-Gallery#2307]
Hi,

In a folder, I can see black thumbnails that lead to photos that no longer exists on my device. See screenshot:
![image](https://user-images.githubusercontent.com/85929121/146780987-a44830f7-75e9-45de-b7bf-27eab194e904.png)

Details:
- I have Android 11.
- Version on a screenshot is 6.22.3 (GitHub build), 6.22.2 also had this issue.
- Last version without an issue was 6.22.1 (also GitHub build).
- I don't remember what I did with those files, probably I did some editing (rotating) with a name change, on version 6.22.1. As you can see on a screenshot, the photo with text has the same file name (but with appended _1) as the next black thumbnail.
- Files doesn't exist on a device. I don't see them neither in Simple File Manager, nor in other file managers, nor in system gallery app, nor when plugged the phone to the computer.
- When I try to open the photo, nothing happens. When I want to see its details or remove, I get a toast with an error, that file doesn't exist.
- Also, I can't run a slideshow in this directory, probably due to these non-existing files.
- It's not related to my settings. I reinstalled the app with default settings and the problem still occurs. Also, clearing cache didn't help.

**Comment by tibbi** [comment]
any way of reproducing the creation of those files?

**Comment by Aga-C** [comment]
I experimented a bit and was able to reproduce it. Actually, now I don't know if this issue belongs here or to Simple File Manager, maybe to both. Here are steps:

1. Open Simple File Manager (I have 6.11.2).
2. Rename any photo file. In my case, I was deleting a few digits at the end of the filename.
3. Open Simple Gallery.

You should be able to see, that there is still an entry with old file name that you can't enter. When I revert to 6.22.1 everything's ok. However, when I change the name with other file manager (in this case Total Commander), Simple Gallery doesn't show the old entry.

**Comment by tibbi** [comment]
ye, it is a file manager issue. Try some other file manager for reproducing it, it should work fine there.

**Comment by Aga-C** [comment]
Yes, but still, gallery shouldn't show invalid entries (as it is in 6.22.1). 

Also regarding file managers:
- Built-in file manager shows these invalid files that were renamed by Simple File Manager. But renaming here works fine.
- Simple File Manager doesn't show invalid entries. Changing name causes this issue.
- Total Commander doesn't show invalid entries, and renaming works well.

**Comment by Aga-C** [comment]
Finally, I was able to find a workaround how to get rid of these invalid files, but it's a bit complicated and not user-friendly. Maybe there are some more details to these ways, but I haven't checked it thoroughly, and I don't want to experiment more with this on my phone. 
Here are the workarounds:
- Some of these non-existing files (the old ones, that I've shown in the first post) I was able to see in Simple File Manager after toggling on showing hidden files. These files were empty, but it was possible to remove them. Note here: I don't have turned on showing hidden files in Simple Gallery.
- Unfortunately, this way wasn't working with the new invalid entries (that I've created today, while trying to reproduce a bug). Simple File Manager doesn't see them. However, I was able to delete them directly in Simple Gallery, but I had to disable the trash bin.

Still, despite fixing this in Simple File Manager, I think that Simple Gallery shouldn't show these files, like it was in 6.22.1.
